packages/nimble-components/src/dialog/tests/dialog.stories.ts

```diff
@@ -16,6 +26,9 @@ interface DialogArgs {
     includeFooterButtons: boolean;
     preventDismiss: boolean;
     content: ExampleContentType;
+    width: DialogSizeOptions;
+    height: DialogSizeOptions;
+    maxHeight: DialogSizeOptions;
```

Comment:
I think showing this as three separate properties is misleading. You wouldn't want to mix small and large tokens. Instead you would use all small or all large tokens and then either height or max-height.

Can you follow a pattern more like Spinner `size` entry in storybook?
I think the states are small fixed, small growable, large fixed, and large growable.
And the usage details should cover targeting the `part(dialog)` CSS Part of the nimble-dialog

Reply:
I added as suggested, but only "small growable" and "large fixed". I don't think the others make sense:
- small fixed: we don't have a small fixed height, when the dialog is small, it is set to "fit-content"
- large growable: we could set the width to be large and the max height to be the large one, and the height to "fit-content".. but then we'll just have a very wide dialog with a small height, that would only grow with the content. That doesn't look like a desired use case..

Let me know your thoughts

Reply:
Adding some more guidance about how the client specifies the size would be helpful. Right now they can toggle between the options and reverse engineer the setting by looking at the "Show code" view but we can be more explicit.

A good pattern to follow is what the table and spinner docs do where they have a dedicated section called Usage Docs >> Sizing ([example](https://nimble.ni.dev/storybook/?path=/docs/components-spinner--docs#sizing)). To do this you'll need to turn off `autodocs` in the story and add a `.mdx` file that controls the structure of the document ([example](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/spinner/tests/spinner.mdx)). In the sizing section you'd describe which design tokens we recommend and how to configure the dialog to use them.

Reply:
Sounds good, thanks for providing the examples! I've added usage docs similarly to the spinner. This is what they look like:

![image](https://github.com/ni/nimble/assets/57374665/969c4291-f78f-4cec-988f-4945da186863)

Let me know if it needs other modifications